test/unit/logic/transaction_pool.js

```diff
@@ -110,27 +170,1073 @@ describe('transactionPool', () => {
+			);
+			transactionPool.transactionInPool.returns(false);
+			done();
+		});
+
+		it('should return error when transaciton is already processed', done => {
+			const transaction = {
+				id: '1311188423423423L',
+				type: transactionTypes.MULTI,
+			};
+			transactionPool.transactionInPool.returns(true);
+			transactionPool.processUnconfirmedTransaction(transaction, true, res => {
+				expect(res).to.deep.eql(
+					'Transaction is already processed: 1311188423423423L'
+				);
+				done();
+			});
+		});
+
+		it('should fail to add transaction to queue when bundled is enabled', done => {
+			const transaction = {
+				id: '1234L',
+				type: transactionTypes.MULTI,
+				bundled: true,
+			};
+			sinon
+				.stub(transactionPool, 'queueTransaction')
+				.callsArgWith(1, 'Failed to queue bundled transaction');
+			transactionPool.processUnconfirmedTransaction(transaction, true, res => {
+				expect(res).to.deep.eql('Failed to queue bundled transaction');
+				done();
+			});
+		});
+
+		it('should add transaction to queue when bundled is enabled', done => {
+			const transaction = {
+				id: '1234L',
+				type: transactionTypes.MULTI,
+				bundled: true,
+			};
+			transactionPool.queueTransaction.callsArgWith(1, transaction);
+			transactionPool.processUnconfirmedTransaction(transaction, true, res => {
+				expect(res).to.deep.eql(transaction);
+				done();
+			});
+		});
+
+		it('should fail to process and verify transaction', done => {
+			const transaction = {
+				id: '1234L',
+				type: transactionTypes.MULTI,
+				bundled: false,
+			};
+
+			const processVerifyTransactionStub = sinon
+				.stub()
+				.callsArgWith(2, 'Failed to process unconfirmed transaction');
+			TransactionPool.__set__(
+				'__private.processVerifyTransaction',
+				processVerifyTransactionStub
+			);
+
+			transactionPool.processUnconfirmedTransaction(transaction, true, res => {
+				expect(res).to.deep.eql('Failed to process unconfirmed transaction');
+				done();
+			});
+		});
+
+		it('should process and verify transaction', done => {
+			const transaction = {
+				id: '1235674L',
+				type: transactionTypes.MULTI,
+				bundled: false,
+			};
+
+			const processVerifyTransactionStub = sinon
+				.stub()
+				.callsArgWith(2, transaction);
+			TransactionPool.__set__(
+				'__private.processVerifyTransaction',
+				processVerifyTransactionStub
+			);
+			transactionPool.processUnconfirmedTransaction(transaction, true, res => {
+				expect(res.id).to.deep.eql(transaction.id);
+				done();
+			});
+		});
+	});
+
+	describe('receiveTransactions', () => {
+		it('should throw error when transaction in invalid', () => {
+			const invalidTransaction = {
+				id: '10431411423423423L',
+				type: transactionTypes.MULTI,
+			};
+			return expect(() => {
+				transactionPool.receiveTransactions(invalidTransaction, {});
+			}).to.throw();
+		});
+
+		it('should receive transaction already processed error when the transaction is processed before', done => {
```

Comment:
Test should validate that `processUnconfirmedTransaction` was called with valid parameters and when returns error or ok